Cloud Build failure: docker build on staging — @tailwindcss/oxide-linux-x64-gnu missing from lockfile (build dba35d3a)

**Type:** build-failure
**Build ID:** `dba35d3a-e66a-489a-9162-72dc71e0733d`
**Console:** https://console.cloud.google.com/cloud-build/builds/dba35d3a-e66a-489a-9162-72dc71e0733d?project=194468397458
**Commit:** [`3646003`](https://github.com/sustainability-software-lab/cal-bioscape-frontend/commit/36460038c01c644033853bc3a6eff953ffa25ff8)

## Summary

`docker build` step on `staging` failed at `npm run build` because `@tailwindcss/oxide-linux-x64-gnu@4.1.18` (the Linux x64 native binary for Tailwind's CSS processing engine) was absent from `node_modules` inside the container.

## Context

- **Branch:** `staging`
- **Trigger:** Push of commit `3646003` ("fix: rename SAF Plants label to Biorefineries")
- **Commit subject:** fix: rename SAF Plants label to Biorefineries (closes #48) (#50)
- **Build duration:** ~37 seconds (failed at `npm run build` inside `docker build`)
- **Build region:** global (not regional)

## Symptom

```
Build error occurred
Error: Turbopack build failed with 2 errors:
./node_modules/mapbox-gl/dist/mapbox-gl.css
Error evaluating Node.js code
Error: Cannot find native binding. npm has a bug related to optional dependencies
(https://github.com/npm/cli/issues/4828). Please try `npm i` again after removing
both package-lock.json and node_modules directory.

Caused by: Error: Cannot find module '@tailwindcss/oxide-linux-x64-gnu'
Caused by: Error: Cannot find module './tailwindcss-oxide.linux-x64-gnu.node'
```

## Root Cause

Commit `a63037a` ("fix: regenerate lockfile to resolve npm ci picomatch conflict and high CVEs") regenerated `package-lock.json` on macOS ARM64. This regeneration recorded only `@tailwindcss/oxide-darwin-arm64` as an installed package entry and omitted `@tailwindcss/oxide-linux-x64-gnu`. When `npm install` runs inside the Linux Docker container, it follows the lockfile and skips the Linux binary, so `@tailwindcss/oxide/index.js` cannot load its native binding and the Turbopack CSS compilation step fails.

This is the same root cause as issue #57 / commit `ae47ac2` which fixed `lightningcss-linux-x64-gnu` with the same pattern. That fix only addressed `lightningcss`; the oxide binary was also missing but didn't surface until after the oxide version involved its native binding in Turbopack's CSS pipeline.

The last successful build on `staging` was `e5f2d7a8` (built commit `56f394b5`, which predated the `a63037a` lockfile regeneration and still had the Linux entry recorded).

## References

- Related prior fix: #57 / commit `ae47ac2` (`lightningcss-linux-x64-gnu`)
- npm optional deps bug: https://github.com/npm/cli/issues/4828
- Failed build console: https://console.cloud.google.com/cloud-build/builds/dba35d3a-e66a-489a-9162-72dc71e0733d?project=194468397458

## Notes

- When future lockfile regenerations are done on macOS, also add any new Linux-specific optional binaries (`*-linux-x64-gnu`) to `optionalDependencies` in `package.json` so they are always recorded in the lockfile regardless of host platform.
- The `musl` variants (`*-linux-x64-musl`) are not needed since the Dockerfile uses `node:20.20.0-slim` (Debian/glibc), not Alpine.

*Filed by Claude Code via `/fix-cloud-build-failure`.*

--- a/package.json
+++ b/package.json
@@ -38,7 +38,8 @@
     "tailwind-merge": "^3.2.0"
   },
   "optionalDependencies": {
-    "lightningcss-linux-x64-gnu": "1.30.2"
+    "lightningcss-linux-x64-gnu": "1.30.2",
+    "@tailwindcss/oxide-linux-x64-gnu": "4.1.18"
   },
   "devDependencies": {
     "@eslint/eslintrc": "^3",

